### PR TITLE
Ts

### PR DESCRIPTION
**Pull request by sangeeta1998**

### CODE REVIEW

File: MiBench/automotive/basicmath/src/basicmath_large.c [hunk 1]
```diff
@@ -89,35 +99,42 @@ int main(void)
...
+  }
+  
+  end = clock();
+  cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
+ 
+
+ printf("\n");
   for (l = 0x3fed0169L; l < 0x3fed4169L; l++)
     {
 	 usqrt(l, &q);
```
Comment (matsbror) [inline MiBench/automotive/basicmath/src/basicmath_large.c:120]: Why is this loop not included in the timing measurement?

File: MiBench/automotive/basicmath/src/basicmath_large.c [hunk 2]
```diff
@@ -89,35 +99,42 @@ int main(void)
...
     printf("%3.0f degrees = %.12f radians\n", X, deg2rad(X));
   puts("");
-/*   for (X = 0.0; X <= (2 * PI + 1e-6); X += (PI / 180)) */
+
   for (X = 0.0; X <= (2 * PI + 1e-6); X += (PI / 5760))
     printf("%.12f radians = %3.0f degrees\n", X, rad2deg(X));
   
-  
+  end = clock();
+  cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
```
Comment (matsbror) [inline MiBench/automotive/basicmath/src/basicmath_large.c:136]: Will this not overwrite the previous assignment to `cpu_time_used`?

File: MiBench/automotive/bitcount/src/bitcnts.c [hunk 3]
```diff
@@ -1,11 +1,3 @@
-/* +++Date last modified: 05-Jul-1997 */
-
```
Comment (matsbror) [inline MiBench/automotive/bitcount/src/bitcnts.c:2]: Why did you remove this?

File: MiBench/automotive/bitcount/src/bitcnts.c [hunk 4]
```diff
@@ -52,42 +46,45 @@ int main(int argc, char *argv[])
...
   };
-  if (argc<2) {
-    fprintf(stderr,"Usage: bitcnts <iterations>\n");
+  if (argc < 2) {
+    fprintf(stderr, "Usage: bitcnts <iterations>\n");
     exit(-1);
-	}
-  iterations=atoi(argv[1]);
-  
+  }
```
Comment (matsbror) [inline MiBench/automotive/bitcount/src/bitcnts.c:52]: It's difficult to review your pull request when you change the formatting of the code that you do not add or change. Please try not to do that.

File: MiBench/consumer/jpeg/djpeg/src/djpeg.c [hunk 5]
```diff
@@ -415,180 +415,143 @@ COM_handler (j_decompress_ptr cinfo)
...
-  /* Insert custom COM marker processor. */
   jpeg_set_marker_processor(&cinfo, JPEG_COM, COM_handler);
 
-  /* Now safe to enable signal catcher. */
 #ifdef NEED_SIGNAL_CATCHER
   enable_signal_catcher((j_common_ptr) &cinfo);
 #endif
 
-  /* Scan command line to find file names. */
-  /* It is convenient to use just one switch-parsing routine, but the switch
```
Comment (matsbror) [inline MiBench/consumer/jpeg/djpeg/src/djpeg.c:457]: Do not remove these and other comments from the original code.

File: MiBench/consumer/jpeg/djpeg/src/djpeg.c [hunk 6]
```diff
@@ -415,180 +415,143 @@ COM_handler (j_decompress_ptr cinfo)
...
   argc = ccommand(&argv);
 #endif
 
   progname = argv[0];
   if (progname == NULL || progname[0] == 0)
-    progname = "djpeg";		/* in case C library doesn't provide it */
+    progname = "djpeg";
+
+  clock_t start, end;
+  start = clock();
```
Comment (matsbror) [inline MiBench/consumer/jpeg/djpeg/src/djpeg.c:440]: I think this is too early for timing. We looked at this yesterday, remember?

File: MiBench/consumer/jpeg/djpeg/src/djpeg.c [hunk 7]
```diff
@@ -415,180 +415,143 @@ COM_handler (j_decompress_ptr cinfo)
...
   } else {
-    /* default output file is stdout */
     output_file = write_stdout();
   }
 
 #ifdef PROGRESS_REPORT
   start_progress_monitor((j_common_ptr) &cinfo, &progress);
 #endif
 
-  /* Specify data source for decompression */
```
Comment (matsbror) [inline MiBench/consumer/jpeg/djpeg/src/djpeg.c:516]: Here is a suitable place to start timing.

File: MiBench/consumer/jpeg/djpeg/src/djpeg.c [hunk 8]
```diff
@@ -415,180 +415,143 @@ COM_handler (j_decompress_ptr cinfo)
...
 #endif
 
-  /* Finish decompression and release memory.
-   * I must do it in this order because output module has allocated memory
-   * of lifespan JPOOL_IMAGE; it needs to finish before releasing memory.
-   */
-  (*dest_mgr->finish_output) (&cinfo, dest_mgr);
+  (*dest_mgr->finish_output)(&cinfo, dest_mgr);
   (void) jpeg_finish_decompress(&cinfo);
   jpeg_destroy_decompress(&cinfo);
```
Comment (matsbror) [inline MiBench/consumer/jpeg/djpeg/src/djpeg.c:554]: Here is a suitable place to stop timing.

File: MiBench/office/stringsearch/src/pbmsrch_large.c [hunk 9]
```diff
@@ -2743,7 +2746,12 @@ NULL};
       for (i = 0; find_strings[i]; i++)
       {
             init_search(find_strings[i]);
+            start=clock();
```
Comment (matsbror) [inline MiBench/office/stringsearch/src/pbmsrch_large.c:2749]: This is a bit tricky. What is init_search doing? If it should be included in the timing, then put the timing outside of the for-loop. Otherwise you need to accumulate the `cpu_times_used`

File: MiBench/security/rijndael/src/aesxam.c [hunk 10]
```diff
@@ -1,311 +1,102 @@
...
- /*
-   -----------------------------------------------------------------------
-   Copyright (c) 2001 Dr Brian Gladman <brg@gladman.uk.net>, Worcester, UK
-   
-   TERMS
-
-   Redistribution and use in source and binary forms, with or without
-   modification, are permitted provided that the following conditions
-   are met:
-   1. Redistributions of source code must retain the above copyright
```
Comment (matsbror) [inline MiBench/security/rijndael/src/aesxam.c:11]: Again, do not remove comments.

File: MiBench/security/rijndael/src/aesxam.c [hunk 11]
```diff
@@ -1,311 +1,102 @@
...
+        printf("The output file: %s could not be opened\n", argv[1]);
+        err = -6;
+        goto exit;
     }
 
-    if(toupper(*argv[3]) == 'E')
-    {                           /* encryption in Cipher Block Chaining mode */
+    if (toupper(*argv[3]) == 'E')
+    { /* encryption in Cipher Block Chaining mode */
         set_key(key, key_len, enc, ctx);
```
Comment (matsbror) [inline MiBench/security/rijndael/src/aesxam.c:77]: should this not be included?

**Review by matsbror — CHANGES_REQUESTED** [review]

File: MiBench/automotive/basicmath/src/basicmath_large.c (hunk 2 above)
Comment (sangeeta1998) [reply MiBench/automotive/basicmath/src/basicmath_large.c:136]: Yes Mats.
I am adding total_cpu_time += cpu_time_used; in order for it to not overlap.

File: MiBench/consumer/jpeg/djpeg/src/djpeg.c (hunk 5 above)
Comment (sangeeta1998) [reply MiBench/consumer/jpeg/djpeg/src/djpeg.c:457]: When I started, I was looking for comments to understand, so for keeping the repo locally I removed the initial headings. Will not do that again, and will modify only in the required places.

File: MiBench/consumer/jpeg/djpeg/src/djpeg.c (hunk 7 above)
Comment (sangeeta1998) [reply MiBench/consumer/jpeg/djpeg/src/djpeg.c:516]: Okay

File: MiBench/office/stringsearch/src/pbmsrch_large.c (hunk 9 above)
Comment (sangeeta1998) [reply MiBench/office/stringsearch/src/pbmsrch_large.c:2749]: I am adding total cpu time to accumulate the cpu_times_used

File: MiBench/security/rijndael/src/aesxam.c (hunk 10 above)
Comment (sangeeta1998) [reply MiBench/security/rijndael/src/aesxam.c:11]: Yes Mats

File: MiBench/security/rijndael/src/aesxam.c (hunk 11 above)
Comment (sangeeta1998) [reply MiBench/security/rijndael/src/aesxam.c:77]: I am not sure, based on argv[3], it is choosing between the encryption or decryption

File: MiBench/office/stringsearch/src/pbmsrch_large.c (hunk 9 above)
Comment (sangeeta1998) [reply MiBench/office/stringsearch/src/pbmsrch_large.c:2749]: Okay

File: MiBench/automotive/basicmath/src/basicmath_large.c (hunk 1 above)
Comment (sangeeta1998) [reply MiBench/automotive/basicmath/src/basicmath_large.c:120]: For the first loop, I could understand the range. But not for the second one, so was confused. I am adding timers for the second loop too.

File: MiBench/automotive/bitcount/src/bitcnts.c (hunk 4 above)
Comment (sangeeta1998) [reply MiBench/automotive/bitcount/src/bitcnts.c:52]: Okay

**Comment by sangeeta1998** [comment]
.
